lib/player.rb

```diff
@@ -12,15 +14,28 @@ def place_ship(x, y, ship_name, orientation)
   end
 
   def attack(cell_position)
-    if !@grid.ship?(cell_position)
-      :water
-    else
-      :hit
-    end
+    ship = @grid.ship_on(cell_position)
+    there?(ship) ? hit(cell_position, ship) : :water
   end
 
   private
 
+  def there?(ship)
```

Comment:
Not sure about this method name, maybe a bit too short! You could say for example `ship_exist`?